src/lib/gundb/FeedStorage.js

```diff
@@ -1064,17 +1073,35 @@ export class FeedStorage {
   async getFromOutbox(event: FeedEvent) {
     let senderPubkey = await this.userStorage.getUserProfilePublickey(get(event, 'data.receiptEvent.from'))
     let recipientPubkey = this.gunuser.is.pub
-    if (senderPubkey) {
-      const data = await this.gun
-        .get(senderPubkey)
-        .get('outbox')
-        .get(recipientPubkey)
-        .get(event.id)
-        .decrypt()
-      log.debug('getFromOutbox:', { id: event.id, recipientPubkey, senderPubkey, data })
-      return data
+
+    if (!senderPubkey) {
+      log.warn('getFromOutbox sender not found:', { event })
+      return
+    }
+
+    log.debug('getFromOutbox sender found:', { event, senderPubkey, recipientPubkey })
+
+    const { data } = event
+    const decryptedData = await this.gun
```

Comment:
yeah, that could have been nice if gun was predictable, you cant await on .on because it might never resolve

Reply:
I've took this into account:
- this.getFromOutbox(updatedFeedEvent) called **without await**
- getFromOutbox updates feed item **if received** data from the onDecrypt

So, we just calling getFromOutbox without waiting it finishes. If we'll got something from GUN - the feed item will be updated 'in backgroun'